File: test/cases/record/test_record_with_procedure.c
#include <stdio.h>
typedef struct {
    int x;
    int y;
} point;
point p;


void resetpoint();

void resetpoint() {
    p.x = 0;
    p.y = 0;

}


int main(void) {
    p.x = 10;
    p.y = 20;
    printf("%d", p.x);
    printf("%d", p.y);
    resetpoint();
    printf("%d", p.x);
    printf("%d", p.y);

    return 0;
}

File: test/cases/record/test_record_with_procedure.pas
program test;
type
  point = record
    x, y: integer
  end;
var
  p: point;

procedure resetPoint;
begin
  p.x := 0;
  p.y := 0
end;

begin
  p.x := 10;
  p.y := 20;
  write(p.x);
  write(p.y);

  resetPoint;

  write(p.x);
  write(p.y)
end.
